pytests/conftest.py

```diff
@@ -0,0 +1,244 @@
+
+    @reraise
+    def post(self, path=None, data=None, json=None, **kwargs):
+        """Adapter for :py:meth:`requests.Session.post`"""
+        return self.session.post(
+            self._join(self.root, (path or '')), data=data, json=json, **kwargs
+        )
+
+    @reraise
+    def put(self, path=None, data=None, **kwargs):
+        """Adapter for :py:meth:`requests.Session.put`"""
+        return self.session.put(
+            self._join(self.root, (path or '')), data=data, **kwargs
+        )
+
+    @reraise
+    def patch(self, path=None, data=None, **kwargs):
+        """Adapter for :py:meth:`requests.Session.patch`"""
+        return self.session.patch(
+            self._join(self.root, (path or '')), data=data, **kwargs
+        )
+
+    @reraise
+    def delete(self, path=None, **kwargs):
+        """Adapter for :py:meth:`requests.Session.delete`"""
+        return self.session.delete(self._join(self.root, (path or '')), **kwargs)
+
+
+class HttpClient(Client):
+    _join = functools.partial(urljoin)
+
+    def __init__(self, host='127.0.0.1', port=None):
+        self.host = host
+        self.port = port
+        super(HttpClient, self).__init__()
+
+    def start(self, wsgi_app, num_workers=1):
+        self._session = requests.Session()
+        self.port = self.port or free_port()
+        self._sock = bjoern.bind_and_listen(
+            self.host,
+            port=self.port,
+            reuse_port=self.reuse_port,
+            listen_backlog=self.listen_backlog,
+        )
+        super(HttpClient, self).start(wsgi_app, num_workers)
+
+    @property
+    def root(self):
+        return urlunsplit(('http', '{}:{}'.format(self.host, self.port), '', '', ''))
+
+
+class UnixClient(Client):
+    _join = functools.partial(posixpath.join)
+
+    def start(self, wsgi_app, num_workers=1):
+        self._session = requests_unixsocket.Session()
+        self._sock = bjoern.bind_and_listen(
+            'unix:{}'.format(self.host),
+            port=self.port,
+            listen_backlog=self.listen_backlog,
+        )
+        super(UnixClient, self).start(wsgi_app, num_workers)
+
+    @property
+    def root(self):
+        return urlunsplit(('http+unix', quote(self.host, safe=''), '', '', ''))
+
+
+@pytest.fixture(params=('HttpClient', 'UnixClient'))
+def client(request):
+    return request.getfixturevalue(request.param.lower())
+
+
+@pytest.fixture
+def httpclient():
+    with HttpClient() as client:
+        yield client
+
+
+@pytest.fixture
+def unixclient():
+    """
+    * don't use pytest's tmpdir/tmp_path as the generated paths easily exceed 108 chars,
+    causing an "OSError: AF_UNIX path too long" on MacOS
+    * also, filenames longer than 64 chars cause urllib hickups, see https://bugs.python.org/issue32958
+    * another thing: tempfile.gettempdir() on MacOS resorts to $TMPDIR
+    which is autogenerated into /var/folders and is also longer than 64 chars
+    """
```

Comment:
btw there's also anonymous sockets with `@...`, but not supported by all platforms. maybe we can test these as well.

Reply:
Good idea! Unfortunately, it looks like `urllib` can't parse the URL, so `requests_unixsocket` can't connect - although it states [the support for abstract unix sockets](https://github.com/msabramo/requests-unixsocket#abstract-namespace-sockets). I have left [a comment with the error message](https://github.com/msabramo/requests-unixsocket/issues/18#issuecomment-517752129), waiting for the author's response. Maybe extract it in a separate task?